src/Injector.php

```diff
@@ -25,12 +25,29 @@
 final class Injector
 {
     private ContainerInterface $container;
+    private bool $cacheReflections = false;
+
+    /**
+     * @var ReflectionClass[]
+     * @psalm-var array<class-string,ReflectionClass>
+     */
+    private array $reflectionsCache = [];
 
     public function __construct(ContainerInterface $container)
     {
         $this->container = $container;
     }
 
+    /**
+     * Caches reflection objects in memory for performance.
```

Comment:
```suggestion
     * Turn on memoization of resolving classes reflections.
```
And add more info about why it needed